forem/forem
# ✂️  Remove search from Connect

## What type of PR is this? (check all applicable)

## Description
This PR is part of the migration process from Elasticsearch to PostgreSQL FTS. **TLDR - this PR _removes_ search from Connect.**

We started by exploring replacing Elasticsearch with PostgreSQL FTS for Connect. In doing so, we learned a few things and decided it would be better to scrap search within Connect all together, for now (the idea of removing search from Connect has product sign-off already).

- The model/object relationships are a bit complicated - they're spread out across `ChatChannelMembership`, `ChatChannel`, `Message`, and `User`.
- Connect itself is rarely used and searching within Connect is used even less.
- The current search in Connect UX is buggy and it's out of the scope of this PR _and_ migration project to fix it right now.
- The current search isn't the most useful as it [searches only the title of the chat and names of the users, not the message contents itself](https://github.com/forem/forem/blob/009c14653ab84da91cf278400350503e056bef29/app/models/chat_channel_membership.rb#L38-L41). In other words, the primary use case is for users who have _many_ chats and want a way to quickly jump to a specific one. As mentioned above, Connect isn't really used.
- Connect is scheduled for a major overhaul - it would be better to revisit search at that time once we know more about what we want the page to look like, what we want to search for, etc.
- This project aims to get Elasticsearch out as quickly and smoothly as possible, which this PR helps address. It's "easier" to remove search than it is to make PostgreSQL FTS performant for something that isn't used and is going to change.

_I'm leaving all the Elasticsearch code for Connect in place for now. We'll remove it at the end in the PR where we take out everything Elasticsearch related._

**Before**
![Screen Shot 2021-04-02 at 3 07 04 PM](https://user-images.githubusercontent.com/15987080/113446105-1bd95380-93c5-11eb-8cde-d191fa178fe7.png)

**After**
![Screen Shot 2021-04-02 at 1 28 14 PM](https://user-images.githubusercontent.com/15987080/113443183-740d5700-93bf-11eb-9d66-db5c38799408.png)

## Related Tickets & Documents
RFC 0153: https://github.com/forem/rfcs/pull/153
https://github.com/orgs/forem/projects/29#card-57639173

## QA Instructions, Screenshots, Recordings
Admittedly, testing Connect in development isn't the smoothest 😅 . Here's what I did:
1. Fire up the app locally (`bin/startup`).
2. Sign up/sign in to your own account.
3. Follow the local admin account (`admin_mcadmin`).
4. Log out of your account and log into the local admin account (info for this is in the seed file).
5. Follow your other account back.
6. Send your account some messages.
7. Make sure everything looks fine, click around.
8. Log out of the local admin account and back into your account.
9. Send the local admin account some messages back.
10. Make sure everything looks fine, click around.

## Added tests?

## [Forem core team only] How will this change be communicated?
- [x] ~I'm not sure how best to communicate this change and need help - **Do we need to include this change in a changelog post or announce this to the community in any way?**~

![finger_cut_gif](https://media.giphy.com/media/3otPoSaQ8DMQeIJZ60/giphy.gif)
